inlong-sort/sort-connectors/kudu/src/main/java/org/apache/inlong/sort/kudu/source/KuduLookupFunction.java

```diff
@@ -0,0 +1,294 @@
+                predicate = KuduPredicate.newComparisonPredicate(column, comparison, (double) value);
+                break;
+            case BOOL:
+                predicate = KuduPredicate.newComparisonPredicate(column, comparison, (boolean) value);
+                break;
+            case UNIXTIME_MICROS:
+                Long time = (Long) value;
+                predicate = KuduPredicate.newComparisonPredicate(column, comparison, time * 1000);
+                break;
+            case BINARY:
+                predicate = KuduPredicate.newComparisonPredicate(column, comparison, (byte[]) value);
+                break;
+            default:
+                throw new IllegalArgumentException("Illegal var type: " + column.getType());
+        }
+        return predicate;
+    }
+
+    public void eval(Object... keys) throws Exception {
+        if (keys.length != keyNames.length) {
+            throw new RuntimeException("The length of lookUpKey and lookUpKeyVals is difference!");
+        }
+        Row keyRow = buildCacheKey(keys);
+        if (this.cache != null) {
+            ConcurrentMap<Row, List<Row>> cacheMap = this.cache.asMap();
+            int keyCount = cacheMap.size();
+            List<Row> cacheRows = this.cache.getIfPresent(keyRow);
+            if (CollectionUtils.isNotEmpty(cacheRows)) {
+                for (Row cacheRow : cacheRows) {
+                    collect(cacheRow);
+                }
+                return;
+            }
+        }
+
+        for (int retry = 1; retry <= maxRetries; retry++) {
+            try {
+                final KuduScanToken.KuduScanTokenBuilder scanTokenBuilder = client.newScanTokenBuilder(table);
+                final Schema kuduTableSchema = table.getSchema();
+                for (int i = 0; i < keyNames.length; i++) {
+                    String keyName = keyNames[i];
+                    Object value = keys[i];
+                    final ColumnSchema column = kuduTableSchema.getColumn(keyName);
+                    KuduPredicate predicate = predicateComparator(column, value);
+                    scanTokenBuilder.addPredicate(predicate);
+                }
+                final List<KuduScanToken> tokenList = scanTokenBuilder.build();
+                ArrayList<Row> rows = new ArrayList<>();
+                for (final KuduScanToken token : tokenList) {
+                    final List<LocatedTablet.Replica> replicas = token.getTablet().getReplicas();
+                    final String[] array = replicas
+                            .stream()
+                            .map(replica -> replica.getRpcHost() + ":" + replica.getRpcPort())
+                            .collect(Collectors.toList()).toArray(new String[replicas.size()]);
+                    final byte[] scanToken = token.serialize();
+                    final KuduScanner scanner = KuduScanToken.deserializeIntoScanner(scanToken, client);
+                    RowResultIterator rowIterator = scanner.nextRows();
```

Comment:
We can use the `KuduScannerIterator` API to process the RowResults through `scanner.iterator()`, which contains keep alive calls to ensure the scanner does not time out.